apache/superset
# fix(Jinja): Extra cache keys to consider vars with set

### SUMMARY
The regex for `ExtraCache` that checks whether the statement has extra cache key calls via `has_extra_cache_key_calls` wasn't considering cases where the callable vars might not be in double curly brackets, such as:

```
{% set customer_id = current_user_email() %}
select '{{customer_id}}' as test;
```

Because of this issue, cached queries for charts would show previously cached data potentially exposing information from other users.

### TESTING INSTRUCTIONS
1. Create a dataset using 
```
{% set customer_id = current_user_email() %}
select '{{customer_id}}' as test;
```
2. Create a table chart based on the dataset
3. Add the chart to a Dashboard
4. Login with another user
5. You should see the current user email

### ADDITIONAL INFORMATION
